app/reducers/Session.js

```diff
@@ -0,0 +1,152 @@
+import Immutable from 'immutable';
+
+import { NEW_SESSION_REQUESTED, NEW_SESSION_BEGAN, NEW_SESSION_DONE,
+        SAVE_SESSION_REQUESTED, SAVE_SESSION_DONE, GET_SAVED_SESSIONS_REQUESTED, GET_SAVED_SESSIONS_DONE,
+        SESSION_LOADING, SESSION_LOADING_DONE,
+        SET_CAPABILITY_PARAM, ADD_CAPABILITY, REMOVE_CAPABILITY, SET_CAPS,
+        SWITCHED_TABS, SAVE_AS_MODAL_REQUESTED, HIDE_SAVE_AS_MODAL_REQUESTED, SET_SAVE_AS_TEXT,
+        DELETE_SAVED_SESSION_REQUESTED, DELETE_SAVED_SESSION_DONE, 
+        CHANGE_SERVER_TYPE, SET_SERVER_PARAM, SET_SERVER,
+        ServerTypes } from '../actions/Session';
+
+// Make sure there's always at least one cap
+const INITIAL_STATE = {
+  savedSessions: [],
+  tabKey: 'new',
+  serverType: ServerTypes.local,
+  server: {
+    local: {},
+    remote: {},
+    sauce: {},
+  },
+  caps: [{
+    type: 'text',
+  }],
+  isCapsDirty: true,
+};
+
+export default function session (state=INITIAL_STATE, action) {
+  switch (action.type) {
+    case NEW_SESSION_REQUESTED:
+      return Immutable.fromJS(state)
```

Comment:
can you motivate the decision to transition to immutable.js instead of sticking with the existing convention?